### PR TITLE
DFA-968 stack for DynamoDB

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
Write a template so the correct database table can be deployed to all environments.

Export table name - use stack name as part of export to namespace the tables.

Update lambdas to import the table name and use it.

See commits.

### CODE REVIEW

File: lambda/dynamo-api/template.yaml [hunk 1]
```diff
@@ -566,6 +574,6 @@ Resources:
     Type: AWS::Logs::LogGroup
 
 Outputs:
-  WebEndpoint:
-    Description: "API Gateway endpoint URL for Prod stage"
-    Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/"
+  ApiBaseUrl:
+    Description: "API base URL - URL used by epxress frontend - supplied as an env var."
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply lambda/dynamo-api/template.yaml:578]: 🦅 👀

File: lambda/dynamo-db/template.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          KeyType: HASH
+        -
+          AttributeName: sk
+          KeyType: RANGE
+
+      BillingMode: PROVISIONED
+
+      ProvisionedThroughput:
+        ReadCapacityUnits: 1
+        WriteCapacityUnits: 1
```
[the remark this answers is not in the document]
Comment (CharlesIC) [reply lambda/dynamo-db/template.yaml:36]: I'd like to use the on-demand mode and avoid specifying the number of read/write units manually. The thing should scale automatically with usage, IMO

**Review by CharlesIC — DISMISSED** [review]
Approved with the note that we'll need to discuss on-demand/autoscaling mode for DynamoDB

File: lambda/dynamo-db/template.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (muthukswamy) [reply lambda/dynamo-db/template.yaml:36]: Example for autoscaling can be found here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#aws-resource-dynamodb-table--examples under ```DynamoDB Table with Application Auto Scaling```

File: backend/dynamo-db/template.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          AttributeType: S
+      KeySchema:
+        -
+          AttributeName: pk
+          KeyType: HASH
+        -
+          AttributeName: sk
+          KeyType: RANGE
+
+      BillingMode: PAY_PER_REQUEST
```
Comment (CharlesIC) [inline backend/dynamo-db/template.yaml:27]: Nice 👍

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**